src/nsqdconnection.js

```diff
@@ -167,9 +167,9 @@ class NSQDConnection extends EventEmitter {
    */
   registerStreamListeners(conn) {
     conn.on('data', data => this.receiveRawData(data));
-    conn.on('end', err => {
+    conn.on('end', () => {
       this.statemachine.goto('CLOSED');
-      this.emit('connection_error', err);
+      this.emit(NSQDConnection.CONNECTION_ERROR, new Error('The other side closed the connection'));
```

Comment:
The thing is. It doesn't get passed anything: https://nodejs.org/api/net.html#net_event_end
So I think this is better than undefined :)